new/src/gamma/Gamma.cpp

```diff
@@ -0,0 +1,265 @@
+/********************************************************************************************
+ * Copyright (C) 2023 'yet Another Gamma Index Tool' Developers.
+ * 
+ * This file is part of 'yet Another Gamma Index Tool'.
+ * 
+ * 'yet Another Gamma Index Tool' is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * 'yet Another Gamma Index Tool' is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with 'yet Another Gamma Index Tool'.  If not, see <http://www.gnu.org/licenses/>.
+ ********************************************************************************************/
+
+#include "Gamma.hpp"
+
+#include <limits>
+#include <cmath>
+
+#include <iostream>
+
+namespace yagit{
+
+namespace{
+const float Nan{std::numeric_limits<float>::quiet_NaN()};
+const float Inf{std::numeric_limits<float>::infinity()};
+}
+
+namespace{
+constexpr float distSq1D(float x1, float x2){
+    return (x2 - x1) * (x2 - x1);
+}
+
+constexpr float distSq2D(float x1, float y1, float x2, float y2){
+    return (x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1);
+}
+
+constexpr float distSq3D(float x1, float y1, float z1, float x2, float y2, float z2){
+    return (x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1) + (z2 - z1) * (z2 - z1);
+}
```

Comment:
Add a comment mentioning that these methods calculate `squared Euclidean distance`.